cms/djangoapps/contentstore/views/course.py

```diff
@@ -519,11 +551,15 @@ def format_library_for_view(library):
             u'can_edit': has_studio_write_access(request.user, library.location.library_key),
         }
 
-    courses_iter = _remove_in_process_courses(courses_iter, in_process_course_actions)
+    courses_iter = _remove_in_process_courses(courses, in_process_course_actions)
     in_process_course_actions = [format_in_process_course_view(uca) for uca in in_process_course_actions]
+    archived_courses_iter = ()
+    if archived_courses:
+        archived_courses_iter = _remove_in_process_courses(archived_courses, in_process_course_actions)
```

Comment:
Instead of modifying the different functions generating the raw course lists, can't we simply split the list here _after_ all other processing has happened?  Instead of all the changes in this file we have now, we would simply have three additional lines of code here (and all other comments in this file would become irrelevant).